drivers/gpio/gpio_ite_it8801.c

```diff
@@ -0,0 +1,486 @@
+	if (ret) {
+		LOG_ERR("Failed to clear bits raw (ret %d)", ret);
+		return ret;
+	}
+
+	return ret;
+}
+
+static int gpio_it8801_port_toggle_bits(const struct device *dev, gpio_port_pins_t pins)
+{
+	const struct gpio_it8801_config *config = dev->config;
+	int ret;
+	uint8_t val, new_val;
+
+	ret = i2c_reg_read_byte_dt(&config->i2c_dev, config->reg_sovr, &val);
+	if (ret) {
+		return ret;
+	}
+	/* Toggle raw bits of GPIO data register */
+	new_val = val ^ pins;
+	if (new_val != val) {
+		ret = i2c_reg_write_byte_dt(&config->i2c_dev, config->reg_sovr, new_val);
+		if (ret) {
+			LOG_ERR("Failed to write toggle value (ret %d)", ret);
+			return ret;
+		}
+	}
+
+	return ret;
+}
+
+static int gpio_it8801_manage_callback(const struct device *dev, struct gpio_callback *callback,
+				       bool set)
+{
+	struct gpio_it8801_data *data = dev->data;
+	int rc;
+
+	rc = gpio_manage_callback(&data->callbacks, callback, set);
+
+	return rc;
+}
+
+void it8801_gpio_alert_handler(const struct device *dev)
+{
+	const struct gpio_it8801_config *config = dev->config;
+	struct gpio_it8801_data *data = dev->data;
+	int ret;
+	gpio_port_value_t value;
+	gpio_port_value_t interrupted_pins_level = 0;
+	uint8_t isr_val, ier_val;
+
+	/*
+	 * When the irq_gpios of the mfd is triggered, if the passed dev
+	 * does not match the dev stored in the data struct of GPIO,
+	 * further execution will be halted.
+	 */
+	if (dev != data->gpio_dev) {
+		return;
+	}
+
+	ret = i2c_reg_read_byte_dt(&config->i2c_dev, config->reg_gpisr, &isr_val);
+	if (ret) {
+		LOG_ERR("Failed to read GPIO interrupt status (ret %d)", ret);
+		return;
+	}
+
+	ret = i2c_reg_read_byte_dt(&config->i2c_dev, config->reg_gpier, &ier_val);
+	if (ret) {
+		LOG_ERR("Failed to read GPIO interrupt pin set (ret %d)", ret);
+		return;
+	}
+
+	gpio_it8801_port_get_raw(dev, &value);
+	interrupted_pins_level |= (value & data->level_high_trig);
+	interrupted_pins_level |= ((~value) & data->level_low_trig);
+
+	if ((isr_val & ier_val) || interrupted_pins_level) {
+		/* Clear pending interrupt */
+		ret = i2c_reg_write_byte_dt(&config->i2c_dev, config->reg_gpisr, isr_val);
+		if (ret) {
+			LOG_ERR("Failed to clear GPIO interrupt (ret %d)", ret);
+			return;
+		}
+
+		gpio_fire_callbacks(&data->callbacks, dev, isr_val);
+		/* Level trigger handler again */
+		it8801_gpio_alert_handler(dev);
+	}
+}
+
+static int gpio_it8801_pin_interrupt_configure(const struct device *dev, gpio_pin_t pin,
+					       enum gpio_int_mode mode, enum gpio_int_trig trig)
+{
+	const struct gpio_it8801_config *config = dev->config;
+	struct gpio_it8801_data *data = dev->data;
+	int ret;
+	uint8_t reg_gpcr = config->reg_gpcr + pin;
+	uint8_t new_value = 0, control;
+	uint8_t mask = BIT(pin);
+
+	if (ioex_check_is_not_valid(dev, pin)) {
+		return -ENOTSUP;
+	}
+
+	/* Disable irq before configuring it  */
+	ret = i2c_reg_update_byte_dt(&config->i2c_dev, config->reg_gpier, mask, new_value);
```

Comment:
can you use `0` here for the last argument and avoid initializing `new_value`?

Reply:
Done.